ui/src/shared/components/CustomTimeRangeDropdown.tsx

```diff
@@ -0,0 +1,89 @@
+import React, {PureComponent} from 'react'
+import moment from 'moment'
+import classnames from 'classnames'
+
+import {ClickOutside} from 'src/shared/components/ClickOutside'
+import CustomTimeRange from 'src/shared/components/CustomTimeRange'
+
+interface State {
+  expanded: boolean
+}
+
+interface Props {
+  timeRange: {
+    upper: string
+    lower: string
+  }
+  onApplyTimeRange: () => void
+}
+
+class CustomTimeRangeDropdown extends PureComponent<Props, State> {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      expanded: false,
+    }
+  }
+
+  public render() {
+    const {timeRange, onApplyTimeRange} = this.props
+
+    const {expanded} = this.state
+
+    return (
+      <ClickOutside onClickOutside={this.handleCloseDropdown}>
```

Comment:
the name for this handler is a bit confusing. it is technically handling the click outside but is doing the closing of the dropdown. Consider renaming to `handleClickOutside`.